### PR TITLE
Add an ACTS package

### PR DESCRIPTION
**Pull request by HadrienG2**
This is a build recipe for the ACTS HEP tracking toolkit.

### CODE REVIEW

File: var/spack/repos/builtin/packages/acts-core/package.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    variant('tests', default=False, description='Build the unit tests')
+    variant('integration_tests', default=False, description='Build the integration tests')
+    variant('digitization', default=False, description='Build the geometric digitization plugin')
+    variant('dd4hep', default=False, description='Build the DD4hep plugin')
+    variant('tgeo', default=False, description='Build the TGeo plugin')
+    variant('json', default=False, description='Build the Json plugin for Json geometry input/output')
+    variant('material', default=False, description='Build the material plugin')
+
+    depends_on('cmake @3.7:', type='build')
+    depends_on('boost @1.62: +program_options +test')
```
Comment (JavierCVilla) [inline var/spack/repos/builtin/packages/acts-core/package.py]: Are these `boost` variants really needed?

File: var/spack/repos/builtin/packages/acts-core/package.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    def cmake_args(self):
+        spec = self.spec
+        cxxstd = spec['root'].variants['cxxstd'].value
+
+        def cmake_variant(cmake_label, spack_variant):
+            enabled = spec.satisfies('+' + spack_variant)
+            return "-DACTS_BUILD_{0}={1}".format(cmake_label, enabled)
+
+        args = [
```
Comment (JavierCVilla) [inline var/spack/repos/builtin/packages/acts-core/package.py]: Do you think it makes sense to add a variant for the identification plugin?  
(`-DACTS_BUILD_IDENTIFICATION_PLUGIN`)

File: var/spack/repos/builtin/packages/acts-core/package.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    variant('integration_tests', default=False, description='Build the integration tests')
+    variant('digitization', default=False, description='Build the geometric digitization plugin')
+    variant('dd4hep', default=False, description='Build the DD4hep plugin')
+    variant('tgeo', default=False, description='Build the TGeo plugin')
+    variant('json', default=False, description='Build the Json plugin for Json geometry input/output')
+    variant('material', default=False, description='Build the material plugin')
+
+    depends_on('cmake @3.7:', type='build')
+    depends_on('boost @1.62: +program_options +test')
+    depends_on('eigen @3.2.9:', type='build')
```
Comment (JavierCVilla) [inline var/spack/repos/builtin/packages/acts-core/package.py]: Have you seen problems finding `eigen` ? I had to add this to the `cmake_args`:

```
args.extend(["-DEIGEN_INCLUDE_DIR=%s" % spec['eigen'].prefix + "/include/eigen3"])
```

File: var/spack/repos/builtin/packages/acts-core/package.py (hunk 1 above)
Comment (HadrienG2) [reply var/spack/repos/builtin/packages/acts-core/package.py]: They are at least used by unit tests, not sure if something else uses them. That being said, the ACTS build system currently doesn't discriminate between build options and [will always fail if they are not present](https://gitlab.cern.ch/acts/acts-core/blob/master/CMakeLists.txt#L54).

File: var/spack/repos/builtin/packages/acts-core/package.py (hunk 3 above)
Comment (HadrienG2) [reply var/spack/repos/builtin/packages/acts-core/package.py]: CMake's Eigen include path detection is... finnicky to say the least. I needed similar things for my local builds, but strangely enough things Just Work out of the box when building using spack. I suspect that Spack must set some CPATH environment somewhere to make this work.

File: var/spack/repos/builtin/packages/acts-core/package.py (hunk 2 above)
Comment (HadrienG2) [reply var/spack/repos/builtin/packages/acts-core/package.py]: I think it does, let me add this...

Comment (HadrienG2) [reply var/spack/repos/builtin/packages/acts-core/package.py]: Done

File: var/spack/repos/builtin/packages/acts-core/package.py (hunk 1 above)
Comment (HadrienG2) [reply var/spack/repos/builtin/packages/acts-core/package.py]: In this sense, they are always needed, though perhaps for the wrong reasons...

**Comment by HadrienG2** [comment]
Regarding choice of versions, I usually tend to only include the latest release(s) of a package in my PRs for new spack packages, because older releases tend to need more build hacks and most HEP experiments tend to stay very close to the current package versions, making old releases rarely needed.

Now, if you tell me that you are still using some of the older releases and have a good reason to stick with them, I don't mind adding support for them of course :wink:

File: var/spack/repos/builtin/packages/acts-core/package.py (hunk 3 above)
Comment (alalazo) [reply var/spack/repos/builtin/packages/acts-core/package.py]: @HadrienG2 FYI, Spack doesn't set CPATH but injects include paths into the command line. See e.g. #10623 and the other PRs referred therein.

**Pull request merged**